### ISSUE TITLE
[Feature] - Similar to soft assertion, we need ability to ignore steps in debug mode when locator is incorrect to resume lengthy E2E tests. 

### ISSUE THREAD

**Issue by Coding-Means-Always-Learning** [microsoft/playwright#24566]
Hello Team,
I am aware we have the concept of [soft assertion](https://playwright.dev/docs/test-assertions#soft-assertions) however the feature i am requesting is for non-assertion statements mainly the statements used to interact with an element in a debug mode

We have broken down our lengthy full E2E test in different modules with in the same spec. However because of shift left and early testing concepts we encounter changes in the attributes and properties of an element which is quite common for an application which is still being developed. 

The challenge however is we are unable to continue or complete the test until we fix / update all the change locators. Hence requesting to think about a feature that allow us to ignore steps when locator is incorrect and allows us to continue the E2E test atleast in debug mode. Sharing an example (which I know is not lengthy )

Assume this is in an E2E.spec.js and we have the following steps in it
Step 1 - Login into a web application
Step 2 - Create a  "New" record 
Step 3 - Change the record status to "In Progress"
Step 4 - Add images to the record
Step 5 - Change the record status to "For Review"
Step 6 - verify the image added and mark record as "Completed"
 
Supposedly if there is a development change for following locators which I would not be aware of until I run the complete test
- "New" record element in Step 2
- "In Progress" element in Step 3
-  "Add image" element in Step 4

And if I happen to run the test even using debug mode, I will not be able to step over to Step 3 without fixing the locator in Step 2. Once I fix locator in Step 2 and will run the test again , the test will fail in Step 3. And I have to repeat the process of fixing and re-running the test. This eats up lot of time and at times we spend more time than manual testing to be able to finish our E2E test

The idea is to have a feature in debug mode which allows me to skip over or ignore the step where playwright is unable to select an element due to incorrect locator . User will then step in and note the incorrect step, carry out the step manually and resume the test without needing to start all over again. Also playwright or in fact any users will not remember all the following steps that have been automated. So by skipping and proceeding they can easily refer to steps in playwright inspector and carry out with actions manually and fix them later in the day.

**Comment by yury-s** [comment]
If Steps 2,3 and 4 are independent and you can safely skip Step 2 before testing Step 3, why not put them in separate tests? E.g. moving common preparation logic to beforeEach/beforeAll etc and put each step into it's own test. In that case, if one of the steps fails, playwright will continue running other tests and in the end you'll get errors only in those steps that were affected by the development changes. On the other hand, if Step 3 depends on Step 2, there is no point in skipping Step 2 when it's broken, as Step 3 is bound to fail in that case because require preconditions are not met.

**Comment by Coding-Means-Always-Learning** [comment]
> If Steps 2,3 and 4 are independent and you can safely skip Step 2 before testing Step 3, why not put them in separate tests? E.g. moving common preparation logic to beforeEach/beforeAll etc and put each step into it's own test. In that case, if one of the steps fails, playwright will continue running other tests and in the end you'll get errors only in those steps that were affected by the development changes. On the other hand, if Step 3 depends on Step 2, there is no point in skipping Step 2 when it's broken, as Step 3 is bound to fail in that case because require preconditions are not met.

Hello @yury-s ,
Thank you for looking into this

Step 3 in here depends on Step so cannot be isolated and you are right if I happen to run the test I should not be able to move to Step 3 before I complete Step 2. Hence, I am asking for a feature in debug mode only so that even if there is a failure it allows us to resume the test. Since the browser window will be headed the action step required to be carried out by Playwright will instead be manually carried out by the user and the user should resume the test through playwright inspector to check if there are any other locator failures. The user will note all such locator issues and update them later in the day or in next sprint cycle. 

This way we could still complete the lengthy E2E test with aid of manual intervention where playwright locator step fails. This will allow tests to complete and help to ensure project delivery is on time without having to worry to fix all the troubling locators at the same time

**Comment by dgozman** [comment]
@Coding-Means-Always-Learning Thank you for the explanation. Unfortunately, this is out of scope for Playwright, which focuses on automated testing only. Closing this issue as not planned.
